src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+
+    private String name = "Isosceles Trapezoid";
+    private String color;
+    private double firstLeg, secondLeg, thirdLeg, height, smallPiece;
+
+    public IsoscelesTrapezoid(double firstLeg, double secondLeg, double thirdLeg, String color) {
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+        this.thirdLeg = thirdLeg;
+        this.color = color;
+    }
+
+    @Override
+    public double findArea() {
+        if (firstLeg > secondLeg) {
+            smallPiece = (firstLeg - secondLeg) / 2;
+        } else {
+            smallPiece = (secondLeg - firstLeg) / 2;
+        }
+        height = Math.sqrt((Math.pow(thirdLeg, 2) - Math.pow(smallPiece, 2)));
+        return ((firstLeg + secondLeg) / 2) * height;
+    }
+
+    @Override
+    public String printAllInformation() {
+        return "Figure" + ": " + getName() + ", area: " + findArea() +
+                " sq. units, firstLeg: " + getFirstLeg() + " units, secondLeg: " + getSecondLeg() +
+                " units, thirdLeg: " + getThirdLeg() + "units, color: " + getColor();
```

Comment:
Use only eng in messages/code. The 'printAllInformation' method concatenates strings directly which may lead to less readable code when dealing with many variables. Consider using String.format for better readability.